### PR TITLE
Fix update sessions when leader change happens

### PR DESCRIPTION
**Pull request by Aiee**
## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### Issue(s) number: 
Close https://github.com/vesoft-inc/nebula-ent/issues/2152, maybe relates to https://github.com/vesoft-inc/nebula-ent/issues/2176

#### Description:
`UpdateSessions()` should deal with errors like leader change.

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by HarrisChu — APPROVED** [review]

**Comment by shanlai** [comment]
graph crash when i check the issue
graph info:
```
I20230110 03:31:20.485074 331532 MetaClient.cpp:3131] Load leader of "192.168.8.202":10015 in 2 space
I20230110 03:31:20.485116 331532 MetaClient.cpp:3131] Load leader of "192.168.8.202":10020 in 2 space
I20230110 03:31:20.485142 331532 MetaClient.cpp:3131] Load leader of "192.168.8.202":10025 in 2 space
I20230110 03:31:20.485155 331532 MetaClient.cpp:3137] Load leader ok
E20230110 03:31:24.075230 331047 MetaClient.cpp:758] Send request to "192.168.8.202":10005, exceed retry limit
E20230110 03:31:24.075510 331047 MetaClient.cpp:759] RpcResponse exception: apache::thrift::transport::TTransportException: Dropping unsent request. Connection closed after: apache::thrift::transport::TTransportException: AsyncSocketException: connect failed, type = Socket not open, errno = 111 (Connection refused): Connection refused
E20230110 03:31:24.075748 331533 GraphSessionManager.cpp:291] Update sessions failed: 0
E20230110 03:31:24.075820 331533 GraphSessionManager.cpp:260] Update sessions failed: RPC failure in MetaClient: apache::thrift::transport::TTransportException: Dropping unsent request. Connection closed after: apache::thrift::transport::TTransportException: AsyncSocketException: connect failed, type = Socket not open, errno = 111 (Connect
F20230110 03:31:24.075868 331533 StatusOr.h:253] Check failed: ok()
E20230110 03:31:24.498812 331531 MetaClient.cpp:758] Send request to "192.168.8.202":10005, exceed retry limit
E20230110 03:31:24.498855 331531 MetaClient.cpp:759] RpcResponse exception: apache::thrift::transport::TTransportException: Dropping unsent request. Connection closed after: apache::thrift::transport::TTransportException: AsyncSocketException: connect failed, type = Socket not open, errno = 111 (Connection refused): Connection refused
E20230110 03:31:24.498910 331532 MetaClient.cpp:192] Heartbeat failed, status:RPC failure in MetaClient: apache::thrift::transport::TTransportException: Dropping unsent request. Connection closed after: apache::thrift::transport::TTransportException: AsyncSocketException: connect failed, type = Socket not open, errno = 111 (Connect
(END)
```
graph core
image

**Review by critical27 — DISMISSED** [review]

File: src/graph/session/GraphSessionManager.cpp [hunk 1]
```diff
@@ -256,10 +256,7 @@ void GraphSessionManager::updateSessionsToMeta() {
   // There may be expired queries, and the
   // expired queries will be killed here.
   auto handleKilledQueries = [this](auto&& resp) {
-    if (!resp.ok()) {
-      LOG(ERROR) << "Update sessions failed: " << resp.status();
-      return Status::Error("Update sessions failed: %s", resp.status().toString().c_str());
-    }
+    DCHECK(resp.ok());
```
Comment (critical27) [inline src/graph/session/GraphSessionManager.cpp:259]: Why do we convince that resp is always valid?

Comment (Aiee) [reply src/graph/session/GraphSessionManager.cpp:259]: https://github.com/vesoft-inc/nebula/pull/5225/files#diff-a9174d4e2b477c826e5e4e23a7bf43ffe74a5853249f4de3aeebf3d95cf70364R286-R288

Here we checked the response from meta service, and this guarantees `handleKilledQueries()` will receive a valid response.

**Review by yixinglu — DISMISSED** [review]

**Comment by Aiee** [comment]
Please hold on review, there are some issues with the log.

File: src/graph/session/GraphSessionManager.cpp (hunk 1 above)
Comment (codesigner) [reply src/graph/session/GraphSessionManager.cpp:259]: Implicit dependent precondition of  one piece of code depends on another piece is not a good design, alghough it do works. Self-contained code is better choice.

Comment (Aiee) [reply src/graph/session/GraphSessionManager.cpp:259]: > Implicit dependent precondition of one piece of code depends on another piece is not a good design, alghough it do works. Self-contained code is better choice.

Makes sense to me. I'll move the check into the lambda functions.

**Review by yixinglu — APPROVED** [review]

**Review by critical27 — APPROVED** [review]

**Pull request merged**